src/data_classifier.py

```diff
@@ -0,0 +1,173 @@
+
+
+def classifier(obs,data_dir):
+    stdstar_list = wifes_calib.ref_fname_lookup.keys()
+
+    # classify each obs
+    bias = []
+    domeflat = []
+    twiflat = []
+    dark = []
+    arc = []
+    wire = []
+    stdstar = {}
+    science = {}
+
+    for ob in obs:
+        fn = data_dir + ob + '.fits'
+        f = pyfits.open(fn)
+        imagetype = f[0].header['IMAGETYP'].upper()
+        obj_name = f[0].header['OBJECT']
+        f.close()
+        #---------------------------
+        # check if it is within a close distance to a standard star
+        # if so, fix the object name to be the good one from the list!
+        try:
+            near_std, std_dist = wifes_calib.find_nearest_stdstar(fn)
+            if std_dist < 100.0:
+                obj_name = near_std
+        except:
+            pass
+        #---------------------------
+        # 1 - bias frames
+        if imagetype == 'BIAS':
+            bias.append(ob)
+        # 2 - quartz flats
+        if imagetype == 'FLAT':
+            domeflat.append(ob)
+        # 3 - twilight flats
+        if imagetype == 'SKYFLAT':
+            twiflat.append(ob)
+        # 4 - dark frames
+        if imagetype == 'DARK':
+            dark.append(ob)
+        # 5 - arc frames
+        if imagetype == 'ARC':
+            arc.append(ob)
+        # 6 - wire frames
+        if imagetype == 'WIRE':
+            wire.append(ob)
+        # 7 - standard star
+        if imagetype == 'STANDARD':
+            # group standard obs together!
+            if obj_name in stdstar.keys():
+                stdstar[obj_name].append(ob)
+            else:
+                stdstar[obj_name] = [ob]
+        
+        # all else are science targets (also consider standar star in imagety = OBJECT)
+        if imagetype == 'OBJECT':
+            if obj_name in stdstar_list:
+                # group standard obs together!
+                if obj_name in stdstar.keys():
+                    stdstar[obj_name].append(ob)
+                else:
+                    stdstar[obj_name] = [ob]
+            else:
+                # group science obs together!
+                if obj_name in science.keys():
+                    science[obj_name].append(ob)
+                else:
+                    science[obj_name] = [ob]
+
+    # #------------------
+    # science dictionay
+
+    sci_obs = []
+
+    for obj_name in science.keys():
+        obs_list = science[obj_name]
+        sci_obs.append({'sci':obs_list, 'sky':[]})
+
+
+    #------------------
+    # stdstars dictionary
+    std_obs = []
+
+    for obj_name in stdstar.keys():
+        obs_list = stdstar[obj_name]
+        std_obs.append({'sci':obs_list, 'name':obj_name,'type':['flux', 'telluric']})
+
+
+    obs_metadata = {
+        'bias' : bias,
+        'domeflat' : domeflat,
+        'twiflat' : twiflat,
+        'dark' : dark,
+        'wire' : wire,
+        'arc'  : arc,
+        'sci'  : sci_obs,
+        'std'  : std_obs}
+    
+    return obs_metadata 
```

Comment:
Looks consistent with what we had before to me.

I take it `science` refers to the stars we were trying to observe, `std` are all standard reference data/stars. The combination is called `obs` for `observation`. That seems reasonable to me. And it's used throughout the codebase, I think.

Reply:
Good! Yes, `science` is the scientific target of the observation i.e., a star, a galaxy, supernova, ... Then `std` is the standard star, this is a well-known star used to calibrate the flux of the science target, and for the subtraction of the telluric lines (i.e., absorption lines produced by water molecules and other traces in the atmosphere that affect the spectra of the science target). I guess these are `obs` because they are not pure calibration files of the camera itself such as the `arcs`, `bias`, `flats`, `darks`, or `wire` frames, but are observations of real sky sources. In any case, this is not any established way of naming them, but as far as it's reasonable and consistent, it's fine for me too.